### DISCUSSION TITLE
Sign in page using dynamic routes

### DISCUSSION THREAD

**Discussion by djayci** [nextauthjs/next-auth#8417] [Help]
Hey Next auth community!

I come to you with a use case I haven't found a solution for online. I'll try my best to explain it:
I got a source code that generates a few different files statically:

/[brand]/[region]/[lang]/{page.tsx}

So through staticPaths it generates a few files on the server, like
/tesco/england/en
/continente/portugal/pt

I'm in need now for a login page, but I don't want that login page to live at the route but rather at the end of the dynamic paths, like
/[brand]/[region]/[lang]/login/{page.tsx}

But as far as I can see the pages:{ signIn: "" } doesn't understand what dynamic paths are so always expects an existing page, so my signIn() calls never point to the right place.

My hacky solution for this was to remove the withAuth wrapper function on my middleware and I'm re-routing all /auth/login APIs to the routes I need, but it feels I'm deviating from nextAuth a lot.

Any ideas?

**Comment by Prominence** [comment]
The same question.
